### ISSUE TITLE
.luc files touched during compilation for no apparent reason

### ISSUE THREAD

**Issue by ysalmon** [u-fischer/luaotfload#40]
I stumbled on a problem where latexmk would endlessly compile the following document
```latex
\documentclass{article}
\usepackage{fontspec}
\setmonofont{Source Code Pro}
\begin{document}
gh
\end{document}
```
The problem does not occur on the first compilation (more precisely : if there is not a .aux file left from a previous compilation) or if ``luaotfload-tool --cache=erase`` is executed just prior to re-compiling.

latexmk reports that
```
Changed files, or newly in use since previous run(s):
      '/opt/yann/texlive/2018/texmf-var/luatex-cache/generic/fonts/otl/sourcecodepro-boldit.luc'
      '/opt/yann/texlive/2018/texmf-var/luatex-cache/generic/fonts/otl/sourcecodepro-regular.luc'
```
and indeed, these files are both modified : their md5 sum changes.

A workaround is to instruct latexmk to ignore changes to .luc files, but the whole things still looks very strange. As seems strange that latexmk seems to not be bothered by this change of .luc file when an .aux file is not present in the compilation directory.

Attached is the log file (it remains the same whether the problem arises or not).
[test.log](https://github.com/lualatex/luaotfload/files/2875106/test.log)

**Comment by zauguin** [comment]
On Linux with an up-to-date TeXLive and luaotfload, I am unable to reproduce this: The sourcecodepro cache files do not change. What kind of system do you use?

**Comment by u-fischer** [comment]
Are the luc also changed if you compile simply with lualatex and not with latexmk?

**Comment by ysalmon** [comment]
After taking a break I remembered that I also have a copy of the Source Code Pro fonts in my $HOME/.fonts folder (directly from github) for use in e.g. Libreoffice. I came to suspect some kind of competition between this and the version from the sourceodepro latex package.

So I deleted the version from .fonts. And did not experience the looping compilation. I added the files back, and it does not loop again !

For now things work, but since I do not think I changed anything else on my system, I remain confused as to why this started to happen some weeks ago (I have had these files for several months if not years), and why it does not come back now.

**Comment by u-fischer** [comment]
The question is probably if the fonts in $HOME are currently in the luaotfload database. What happens if you force a recreation of the database by e.g. trying to use an non-existing font?

**Comment by ysalmon** [comment]
Yesterday, forcing recreation of the database in the manner you suggested did not cause the problem to trigger again. And it still does not happen on the MWE above.

However, one of my "real" files started to exhibit the behaviour today (when at some point yesterday I swear it did not), albeit with sourcecodepro-regular.luc only and not the boldit variant. This allows me to answer a previous question : yes, the .luc file gets modified even if latexmk is not involved.

I can also confirm, because lualatex says it at the end of compilation, that it gets the font from my $HOME/.fonts. Now you will be asking : for the MWE that does not exhibit the problem anymore, where do the fonts come from. They come from $HOME/.fonts as well.

**Comment by zauguin** [comment]
@ysalmon Can you upload a copy of the affected `.lua` and `.luc` file before and after a lualatex run?

**Comment by ysalmon** [comment]
Here before and after a single lualatex compilation.

[before.zip](https://github.com/u-fischer/luaotfload/files/2885390/before.zip)
[after.zip](https://github.com/u-fischer/luaotfload/files/2885393/after.zip)

**Comment by ysalmon** [comment]
I tried again to delete the Source Code Pro in my $HOME/.fonts to see what would happen. Compilation was not possible due to the sourcecodepro latex fonts not containing semibold version (apparently), which I now remember was the reason I downloaded a more complete version.

So I did the converse : removed the sourcecodepro latex package and installed the font back in $HOME/.fonts. At first lualatex could only find some variants of the font (not italic, for example). After ``luaotfload-tool -u``, thing got in order, and the .luc files are not uselessly regenerated (for now ?).
